bot/exts/utils/reminders.py

```diff
@@ -168,13 +196,35 @@ async def _reschedule_reminder(self, reminder: dict) -> None:
         log.trace(f"Scheduling new task #{reminder['id']}")
         self.schedule_reminder(reminder)
 
-    @lock_arg(LOCK_NAMESPACE, "reminder", itemgetter("id"), raise_error=True)
     async def send_reminder(self, reminder: dict, expected_time: datetime = None) -> None:
-        """Send the reminder."""
-        is_valid, user, channel = self.ensure_valid_reminder(reminder)
-        if not is_valid:
-            # No need to cancel the task too; it'll simply be done once this coroutine returns.
-            return
+        """Validate reminder, and call sender."""
+        while True:
+            is_valid, abort, user, channel = await self.ensure_valid_reminder(reminder)
+
+            if not is_valid and not abort:
+                # Failed to validate reminder's user, try again in an hour
+                reminder["failures"] += 1
+                await asyncio.sleep(60 * 60)
+
+            elif abort:
+                # Reminder could not be validated after multiple attempts, giving up.
+                # No need to cancel the task too; it'll simply be done once this coroutine returns.
+                return
+            else:
+                # Reminder is valid, continue.
+                break
+
+        await self._format_send_reminder(reminder, expected_time, user, channel)
+
+    @lock_arg(LOCK_NAMESPACE, "reminder", itemgetter("id"), raise_error=True)
+    async def _format_send_reminder(
```

Comment:
Can anyone think of a better name?

Also, putting the underscore prefix is unnecessary in cogs. We can work under the assumption that cogs generally will not have their methods accessed externally because the classes aren't meant to be instantiated directly, and there are very few exceptions where `get_cog` is used to get to a method. I realise the underscore is prefix is used inconsistently in cogs, but I suppose if that is to be changed then it needs to start being brought up in reviews.